### PR TITLE
[CognitiveServices] LUIS Programmatic SDK

### PR DESCRIPTION
**Pull request by pcostantini**
Based on the following Spec PRs:
 - https://github.com/Azure/azure-rest-api-specs/pull/2083
 - https://github.com/Azure/azure-rest-api-specs/pull/2152
 - https://github.com/Azure/azure-rest-api-specs/pull/2214

This PR includes:
 - C# Client for Programmatic API Endpoint generated with AutoRest ([config](https://github.com/Azure/azure-rest-api-specs/tree/current/specification/cognitiveservices/data-plane/LUIS/Programmatic#programmatic-20---csharp-settings))
 - API Token Credential
 - Unit Tests
 - Directory re-structure for CS Language SDKs

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/CognitiveServices/dataPlane/Language/LUIS/Programmatic/Microsoft.Azure.CognitiveServices.LUIS.Programmatic.csproj [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.reference.props'))" />
```
Comment (shahabhijeet) [inline src/SDKs/CognitiveServices/dataPlane/Language/LUIS/Programmatic/Microsoft.Azure.CognitiveServices.LUIS.Programmatic.csproj]: @pcostantini include the comment around the imports of the project file similar to below
https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/CognitiveServices/dataPlane/Search/BingCustomSearch/BingCustomSearch/Microsoft.Azure.CognitiveServices.Search.CustomSearch.csproj#L2

Also include imports that are missing at the bottom of the project file.

File: src/SDKs/CognitiveServices/dataPlane/Language/LUIS/Programmatic/Microsoft.Azure.CognitiveServices.LUIS.Programmatic.csproj [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  <PropertyGroup>
+    <PackageId>Microsoft.Azure.CognitiveServices.Language.LUIS.Programmatic</PackageId>
+    <Description>Provides API functions for consuming the Microsoft Azure Cognitive Services LUIS Programmatic API.</Description>
+    <VersionPrefix>2.0.0</VersionPrefix>
```
Comment (shahabhijeet) [inline src/SDKs/CognitiveServices/dataPlane/Language/LUIS/Programmatic/Microsoft.Azure.CognitiveServices.LUIS.Programmatic.csproj]: remove line 7,8 and replace < VersionPrefix to < Version >

File: src/SDKs/CognitiveServices/dataPlane/Language/LUIS/Programmatic.Tests/Microsoft.Azure.CognitiveServices.LUIS.Programmatic.Tests.csproj [hunk 3]
```diff
@@ -0,0 +1,24 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.test.reference.props'))" />
+  <PropertyGroup>
+    <Description>Microsoft.Azure.CognitiveServices.LUIS-Programmatic.Tests Class Library</Description>
+    <AssemblyName>Microsoft.Azure.CognitiveServices.LUIS-Programmatic.Tests</AssemblyName>
+    <VersionPrefix>1.0.0-preview</VersionPrefix>
+  </PropertyGroup>
+  <PropertyGroup>
+    <TargetFramework>netcoreapp1.1</TargetFramework>
+    <IsPackable>false</IsPackable>
```
Comment (shahabhijeet) [inline src/SDKs/CognitiveServices/dataPlane/Language/LUIS/Programmatic.Tests/Microsoft.Azure.CognitiveServices.LUIS.Programmatic.Tests.csproj]: remove line 10

File: src/SDKs/CognitiveServices/dataPlane/Language/LUIS/Runtime.Tests/Microsoft.Azure.CognitiveServices.LUIS.Tests.csproj [hunk 4]
```diff
@@ -16,7 +16,7 @@
   </ItemGroup>
 
   <ItemGroup>
-    <ProjectReference Include="..\LUIS-Runtime\Microsoft.Azure.CognitiveServices.LUIS.csproj" />
+    <ProjectReference Include="..\Runtime\Microsoft.Azure.CognitiveServices.LUIS.csproj" />
```
Comment (shahabhijeet) [inline src/SDKs/CognitiveServices/dataPlane/Language/LUIS/Runtime.Tests/Microsoft.Azure.CognitiveServices.LUIS.Tests.csproj]: include the imports in LUIS.Tests project.
Remove lines 5,6
Remove lines 9-16 (all these references are implicitly added to each test project)
Add relevant properties about assemblyname, version etc to LUIS.Test project file (basically model this project file according to other test project files.

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
For Azure.CognitiveServices.LUIS.csproj 
1) Add header/footer imports in the project file, include comments around the imports.
For CognitivieServices.Language.tests.csproj
1) Remove lines 10, 25-27
2) Change < VersionPrefix > to < Version >

For Microsoft.Azure.CognitiveServices.Language.csproj 
Model this project according to the comments given for other SDK proejcts.
Basically make test project and SDK projects similar to other data plane projects that are within cognitive services directory (take an example from search sdks)

**Comment by pcostantini** [comment]
Hi @shahabhijeet,
Feedback was applied. Let me know if there is anything else we should look into.
Regards,
-PabloC.

**Review by shahabhijeet — APPROVED** [review]

**Pull request merged**